src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricSerializer.cs

```diff
@@ -111,106 +147,72 @@ private string FormatDouble(double min)
 
         private void WriteTimestamp(StringBuilder sb, DateTime timestamp)
         {
-            sb.Append($" {new DateTimeOffset(timestamp).ToUniversalTime().ToUnixTimeMilliseconds()}");
+            sb.Append($" {new DateTimeOffset(timestamp.ToLocalTime()).ToUnixTimeMilliseconds()}");
```

Comment:
I was quite confused by this for a long time and spent quite some minute spelunking in the microsoft documentation. I believe the root of the issue is that `ToUnixTimeMilliseconds` operates on `DateTimeOffset`, not on `DateTime`, and converting between `DateTime` and `DateTimeOffset` is fraught with timezone conversion nightmares relating to your locale. Unfortunately, there doesn't seem to be a good direct conversion from `DateTime` in `C#` which is endlessly frustrating to me. Unfortunately, I couldn't find a way which I thought was any better so I think this will have to do.

Reply:
Oh, yeah, that makes sense. One alternative I found was to subtract the `DateTime` value for 01/01/1970 from the date, which should result in the same (Unix) timestamp. However, this will also work on the local time and does not change the problem at hand that it is not easily possible to create meaningful timestamps.